### PR TITLE
core/validatorapi: add support for DutyAggregator to validatorapi

### PR DESCRIPTION
**Pull request by dB2510**
Adds support for DutyAggregator to validatorapi. Also adds signature verification for SignedAttestationAndProof.

category: feature
ticket: #1094

### CODE REVIEW

File: core/validatorapi/validatorapi.go [hunk 1]
```diff
@@ -666,6 +671,93 @@ func (c Component) SubmitBeaconCommitteeSubscriptionsV2(ctx context.Context, sub
...
+		if !ok {
+			return errors.New("slot signature not found")
+		}
+
+		// Override selection proof with slot signature from subscription.
+		agg.Message.SelectionProof = sub.SlotSignature
+	}
+
+	psigsBySlot := make(map[eth2p0.Slot]core.ParSignedDataSet)
+	for _, agg := range aggregateAndProofs {
```
Comment (corverroos) [inline core/validatorapi/validatorapi.go:721]: another loop over same slice, can we combine with above maybe?

File: core/validatorapi/validatorapi.go [hunk 2]
```diff
@@ -666,6 +671,93 @@ func (c Component) SubmitBeaconCommitteeSubscriptionsV2(ctx context.Context, sub
...
+
+		// Verify Signature.
+		err = c.verifyPartialSig(pk, func(pubshare *bls_sig.PublicKey) error {
+			return signing.VerifyAggregateAndProof(ctx, c.eth2Cl, pubshare, agg)
+		})
+		if err != nil {
+			return err
+		}
+
+		_, ok := psigsBySlot[agg.Message.Aggregate.Data.Slot]
```
Comment (corverroos) [inline core/validatorapi/validatorapi.go:740]: I'd define a `slot := psigsBySlot[agg.Message.Aggregate.Data.Slot` variable to be less verbose

File: core/validatorapi/validatorapi.go [hunk 3]
```diff
@@ -666,6 +671,93 @@ func (c Component) SubmitBeaconCommitteeSubscriptionsV2(ctx context.Context, sub
...
 }
 
+// AggregateAttestation fetches the aggregate attestation given an attestation.
+// Note: It queries aggregated attestation from DutyDB (this is blocking).
+func (c Component) AggregateAttestation(ctx context.Context, slot eth2p0.Slot, attestationDataRoot eth2p0.Root) (*eth2p0.Attestation, error) {
+	return c.awaitAggAttFunc(ctx, int64(slot), attestationDataRoot)
+}
+
+// SubmitAggregateAttestations receives partially signed aggregateAndProofs.
+func (c Component) SubmitAggregateAttestations(ctx context.Context, aggregateAndProofs []*eth2p0.SignedAggregateAndProof) error {
```
Comment (corverroos) [inline core/validatorapi/validatorapi.go:681]: maybe add an overview of the steps this function does to the godoc

**Review by corverroos — APPROVED** [review]

File: core/validatorapi/validatorapi.go (hunk 1 above)
Comment (dB2510) [reply core/validatorapi/validatorapi.go]: yeah sure

**Pull request merged**